services/templates.ts

```diff
@@ -266,3 +278,93 @@ export const formatTemplatesWithPriceData = (
     ...template,
     lowestPrice: lowestPrices[template.template_id] || '',
   }));
+
+/***
+ * Function returns templates with the following added keys: (used primarily for diaplying user's owned assets in My NFT page)
+ *    totalAssets: Total number of assets owned by 'owner'
+ *    assetsForSale: Number of assets for sale by 'owner'
+ *    lowestPrice: Lowest price of an asset for sale in marketplace
+ * @param {string} owner Owner of assets to look up
+ * @param {number} page  Reference for pagination if number of template categories (based on owned assets) is greater than number of templates displayed per page
+ * @return {Template[]}
+ */
+
+export const getTemplatesWithUserAssetCount = async (
```

Comment:
Can we break this function down with helper functions? Even though everything is named with clear intentions, it's a little confusing to follow step by step.